libarchive/archive_read_support_filter_gzip.c

```diff
@@ -362,6 +395,7 @@ consume_trailer(struct archive_read_filter *self)
 		return (ARCHIVE_FATAL);
 
 	/* XXX TODO: Verify the length and CRC. */
+	state->file_size = archive_le32dec(p + 4);
```

Comment:
Is there any point storing the `file_size` in `state` here?  I think the only useful thing to do here might be to compare the file size with the number of bytes that were actually decompressed and fail if they don't match (modulo 2^32, of course).

Reply:
would you prefer we fail or warn ?  i don't know what the general approach is with libarchive in this space.